### ISSUE TITLE
Support for streams

### ISSUE THREAD

**Issue by Mohamed5550** [PlatformCommunity/flysystem-bunnycdn#79]
Hi 
Thanks for the great adapter!

I wonder if there is support for streams somewhere in the package

I want to use the adapter to upload the video directly to stream videos instead of storage

Thanks in advance

**Comment by sifex** [comment]
Hey @Mohamed5550 , I believe the streaming API is different to the Storage API for bunny, you could use this lib here instead? 

https://github.com/cp6/BunnyCDN-API?tab=readme-ov-file#video

tbh Haven't had much exposure to it

**Comment by Mohamed5550** [comment]
Hi @sifex 
It's a great package, I've used it in my application

But is there any way to make an adapter for it to integrate with laravel filesystems?

**Comment by sifex** [comment]
Yes, please check the README under Laravel, should provide you all you need to know

**Comment by sifex** [comment]
Yes — streams are fully supported. `readStream()` returns a real stream resource via `BunnyCDNClient::stream()` (Guzzle `stream => true` + `detach()`), and `writeStream()` passes resources straight to the upload request, which sets `Content-Length` from `fstat` so large files stream without loading into memory. Note this adapter targets Bunny **Storage Zones** only — Bunny's separate "Stream" video product (`video.bunnycdn.com`) is a different API and out of scope here. For videos, upload via `writeStream` and serve through a pull zone, or call Bunny's Stream API directly.
